lib/user/lib/src/models/user.dart

```diff
@@ -22,41 +21,41 @@ class AppUser {
 
   final TypeOfUser typeOfUser;
 
-  final String referralLink, referredBy;
+  final String? referralLink, referredBy;
   final int referralScore;
 
   final StateEnum state;
-  final String reminderTime;
-  final List<String> notificationTokens;
-  final bool blackboardNotifications;
-  final bool commentsNotifications;
+  final String? reminderTime;
+  final List<String?> notificationTokens;
+  final bool? blackboardNotifications;
+  final bool? commentsNotifications;
   final UserSettings userSettings;
   final UserTipData userTipData;
-  final DateTime createdOn;
-  final Features features;
+  final DateTime? createdOn;
+  final Features? features;
 
   AppUser._({
-    @required this.id,
-    @required this.name,
-    @required this.abbreviation,
-    @required this.typeOfUser,
-    @required this.referralLink,
-    @required this.referredBy,
-    @required this.referralScore,
-    @required this.reminderTime,
-    @required this.notificationTokens,
-    @required this.state,
-    @required this.blackboardNotifications,
-    @required this.commentsNotifications,
-    @required this.userSettings,
-    @required this.userTipData,
-    @required this.createdOn,
+    required this.id,
+    required this.name,
+    required this.abbreviation,
+    required this.typeOfUser,
+    required this.referralLink,
+    required this.referredBy,
+    required this.referralScore,
+    required this.reminderTime,
+    required this.notificationTokens,
+    required this.state,
+    required this.blackboardNotifications,
+    required this.commentsNotifications,
+    required this.userSettings,
+    required this.userTipData,
+    required this.createdOn,
     this.features,
   });
 
-  factory AppUser.create({String id}) {
+  factory AppUser.create({String? id}) {
     return AppUser._(
-      id: id,
+      id: id ?? "Id",
```

Comment:
I don't think `id` should be nullable. I see that `AppUser.create` was used in several places as for a placeholder value (and thus didn't pass `id`) but I think the right thing to do is to make `id` required and non-nullable and just pass a placeholder id in the aforementioned places.